locator/topology.hh

```diff
@@ -0,0 +1,124 @@
+/*
+ *
+ * Modified by ScyllaDB
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: (AGPL-3.0-or-later and Apache-2.0)
+ */
+
+#pragma once
+
+#include <unordered_set>
+#include <unordered_map>
+
+#include <seastar/core/future.hh>
+#include <seastar/core/sstring.hh>
+#include <seastar/util/bool_class.hh>
+
+#include "locator/types.hh"
+
+using namespace seastar;
+
+namespace locator {
+
+class topology {
+public:
+    using inet_address = gms::inet_address;
```

Comment:
nit: this thing appeared during move, but there's namespace locator -wide declaration under the same name. Is this really required here?

Reply:
It previously relied on https://github.com/scylladb/scylladb/blob/d85f731478211c3952114247807eb424bb400b62/locator/token_metadata.hh#L37
I can move it too to locator/types.hh  but I don't like it very much, hence I restricted it to class topology.

Reply:
Funny thing, is that I actually did do that eventually and kept both.
So I'll remove this one.  Should be unneeded.